list-ops/example.py

```diff
@@ -0,0 +1,27 @@
+
+def map_src(function, xs):
+    return [function(elem) for elem in xs]
+
+
+def reduce_src(function, xs, start=None):
+    it = iter(xs)
+    if start is None:
+        try:
+            start = next(it)
+        except:
+            raise TypeError("reduce_src() on empty sequence")
+    accumulator = start
+    for elem in it:
+        accumulator = function(accumulator, elem)
+    return accumulator
+
+
+def length_src(xs):
+    if not xs:
+        return 0
+    else:
+        count = 0
+        start = xs[0]
```

Comment:
This variable is never used and could be problem when the input doesn't support indexing.

This bug would have probably been caught by a linting tool like pylint.